src/fides/api/models/property.py

```diff
@@ -16,10 +25,118 @@ class Property(Base):
     This class serves as a model for digital properties, such as websites or other online platforms.
     """
 
+    def generate_id(self) -> str:
+        """
+        Generate a unique ID in the format 'FDS-XXXXXX' using uppercase alphanumeric characters.
+        """
+        characters = string.ascii_uppercase + string.digits
+        return "FDS-" + "".join(random.choices(characters, k=6))
+
     @declared_attr
     def __tablename__(self) -> str:
         return "plus_property"
 
-    key = Column(String, index=True, nullable=False, unique=True)
+    id = Column(
+        String,
+        primary_key=True,
+        default=generate_id,
+    )
     name = Column(String, nullable=False, unique=True)
     type = Column(EnumColumn(PropertyType), nullable=False)
+
+    experiences: RelationshipProperty[List[PrivacyExperienceConfig]] = relationship(
+        "PrivacyExperienceConfig",
+        secondary="plus_privacy_experience_config_property",
+        back_populates="properties",
+        lazy="selectin",
+    )
+
+    @classmethod
+    def create(
+        cls: Type[Property],
+        db: Session,
+        *,
+        data: dict[str, Any],
+        check_name: bool = True,
+    ) -> Property:
+        experiences = data.pop("experiences", [])
+        prop: Property = super().create(db=db, data=data, check_name=check_name)
+        link_experience_configs_to_property(
+            db, experience_configs=experiences, prop=prop
+        )
+        return prop
+
+    def update(self, db: Session, *, data: Dict[str, Any]) -> Property:
+        experiences = data.pop("experiences", [])
+        super().update(db=db, data=data)
+        link_experience_configs_to_property(
+            db, experience_configs=experiences, prop=self
+        )
+        return self
+
+
+class PrivacyExperienceConfigProperty(Base):
+    @declared_attr
+    def __tablename__(self) -> str:
+        return "plus_privacy_experience_config_property"
+
+    def generate_uuid(self) -> str:
+        """
+        Generates a uuid with a prefix based on the tablename to be used as the
+        record's ID value
+        """
+        try:
+            prefix = f"{self.current_column.table.name[:3]}_"  # type: ignore
+        except AttributeError:
+            prefix = ""
+        uuid = str(uuid4())
+        return f"{prefix}{uuid}"
+
+    # Overrides Base.id so this is not a primary key.
+    # Instead, we have a composite PK of privacy_experience_config_id and property_id
+    id = Column(String(255), default=generate_uuid)
+
+    privacy_experience_config_id = Column(
+        String,
+        ForeignKey("privacyexperienceconfig.id", ondelete="CASCADE"),
+        index=True,
+        nullable=False,
+        primary_key=True,
+    )
+    property_id = Column(
+        String,
+        ForeignKey("plus_property.id", ondelete="CASCADE"),
+        index=True,
+        nullable=False,
+        primary_key=True,
+    )
+
+
+def link_experience_configs_to_property(
```

Comment:
i won't reiterate what i'm trying to say in [this comment on the plus PR](https://github.com/ethyca/fidesplus/pull/1335#discussion_r1519111655) - but, as you know, i don't love how we're establishing a circular dependency here! i do think there are some reasonable ways to change how this is done while keeping almost all of your logic and more substantive implementation details in tact. happy to talk through it more

Reply:
I'll address this in the next sprint to make sure this current iteration goes out on time.